javascript/javascript3/week3/user.js

```diff
@@ -0,0 +1,233 @@
+class Users {
+    constructor() {
+        this.form = document.querySelector('form');
+        this.formLog = document.querySelector('.formLog');
+        this.email = document.querySelector('[type="email"]');
+        this.password = document.querySelector('.password');
+        this.emailLogIn = document.querySelector('.emailLogIn');
+        this.passwordLogIn = document.querySelector('.passwordLogIn');
+        this.inputs = document.querySelectorAll('input');
+        this.fullCollection = document.querySelector('.fullCollection');
+        this.deleteInput = document.querySelector('.deleteInputUser'); 
+        this.url = 'https://crudcrud.com/api/ce64f626ce31465592a412be09c55354/users';
+    }
+
+    clearInputs() {
+        this.inputs.forEach(item => {
+            item.value = '';
+        }); 
+    }
+
+    checkMailInputs(){
+        this.email.addEventListener('keypress', function(e) {
+            if (e.key.match(/[^a-z 0-9 @ \.]/ig)) {
+                e.preventDefault();
+            }
+        });
+    }
+
+    async getResource(url){
+        let response = await fetch(url, {
+            method: 'GET'	
+        });
+        if (!response.ok) {
+            throw new Error(`Could not fetch ${url}, status: ${response.status}`);
+        }
+    
+        return await response.json();
+    }
+
+    async deleteResource(url){
+        await fetch(url, {
+            method: 'DELETE'	
+        });
+        return null;
+    }
+
+    async postData(url, data) {
+        let res = await fetch(url, {
+            method: 'POST',
+            body: JSON.stringify(data),
+            headers: {
+                'Content-type': 'application/json',
+            }
+        });
+    
+        return await res.json();
+    }
+
+    singUp() {
+
+        this.checkMailInputs();
+
+        this.form.addEventListener('submit', (e) => {
+            e.preventDefault();
+
+            if (this.email.value !== '' && this.password.value !== '') {
```

Comment:
You could also add required attribute to the input to take care of this.

Reply:
Thank you for review. I'll try to do something with Users class))). What do you think about Screenshots class?